Source/Core/Core/WatchDog.h

```diff
@@ -0,0 +1,113 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "Common/CommonTypes.h"
+
+#include <compare>
+#include <deque>
+#include <mutex>
+#include <string_view>
+#include <unordered_map>
+#include <vector>
+
+#include <fmt/format.h>
+
+namespace WatchDog
```

Comment:
"WatchDog" seems like a strange name.  What about something like "EventProfiler"

Reply:
Yes, watchdog has a rather particular meaning that doesn't match what this is doing.